pkg/internalserver/signal_handler.go

```diff
@@ -0,0 +1,62 @@
+package internalserver
+
+import (
+	"os"
+	"os/signal"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+)
+
+func newSignalHandler(shutdownDelay time.Duration, logger log.Logger) *serverSignalHandler {
+	return &serverSignalHandler{
+		quit:          make(chan struct{}),
+		shutdownDelay: shutdownDelay,
+		logger:        logger,
+	}
+}
+
+type serverSignalHandler struct {
+	quit          chan struct{}
+	ready         atomic.Bool
+	shutdownDelay time.Duration
+	logger        log.Logger
+}
+
+func (dh *serverSignalHandler) Loop() {
```

Comment:
Why is this called `dh`?

Reply:
Probably a typo since s is next to d :joy:  https://github.com/grafana/mimir-proxies/pull/79/commits/cda305e962a87e44032aa8fa8453198050904bc7